### PR TITLE
feat: add telemetry for external provisioners

### LINKED ISSUE #9556: external provisioners: report usage in telemetry
Would love to see via telemetry what deployments are using external provisioners. I don't have strong thoughts on how we show this, perhaps we submit provisioners to telemetry

### PR DESCRIPTION
**Pull request by coadler**
No telemetry server PR yet.

### CODE REVIEW

**Comment by coadler** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10322** <a href="https://app.graphite.dev/github/pr/coder/coder/10322" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/coder/coder/10322?utm_source=stack-comment).

File: coderd/telemetry/telemetry.go [hunk 1]
```diff
@@ -698,27 +699,46 @@ func ConvertWorkspaceProxy(proxy database.WorkspaceProxy) WorkspaceProxy {
 	}
 }
 
+func ConvertExternalProvisioner(id uuid.UUID, tags map[string]string, provisioners []database.ProvisionerType) ExternalProvisioner {
+	hashedTags := make(map[string]string, len(tags))
+	for k, v := range tags {
+		hash := sha256.Sum256([]byte(v))
+		hashedTags[k] = hex.EncodeToString(hash[:])
```
Comment (ammario) [inline coderd/telemetry/telemetry.go:706]: we don't obscure workspace or template names, so I don't think we should obscure these.

**Review by ammario — APPROVED** [review]
There will have to be corresponding changes in the telemetry-server before this works. It's a bit janky so I can do it for you if you want.

**Comment by coadler** [comment]
> There will have to be corresponding changes in the telemetry-server before this works. It's a bit janky so I can do it for you if you want.

I'll have a PR up after I merge, although I'm mostly just copying what you did for CLI telemetry

**Pull request merged**